### PR TITLE
ByteArrayLoad for ssplit

### PR DESCRIPTION
**Pull request by jerinphilip**
ssplit-cpp half of ByteArray loading the file. Will do the bergamot-translator bits once this reaches consensus.

https://github.com/browsermt/bergamot-translator/issues/104

Added a new workflow file to combinatorially blow-up the matrix. Test scripts are now in a shell-script. The redundant files may be discarded.

### CODE REVIEW

File: src/ssplit/ssplit.cpp [hunk 1]
```diff
@@ -4,20 +4,52 @@
...
+};
+struct imemstream: virtual membuf, std::istream {
+    imemstream(char const* base, size_t size)
+        : membuf(base, size)
+        , std::istream(static_cast<std::streambuf*>(this)) {
+    }
+};
+
+
+}
```
Comment (qianqianzhu) [inline src/ssplit/ssplit.cpp]: Shall we add `// Anonymous namespace` here? To be clearer that the anonymous namespace ends up here.

**Comment by ugermann** [comment]
By and large looks good to me. The detour through in-memory streams involves some unnecessary copying of data. ssplit.cpp already has a buffer-based line scanner that doesn't involve copying data into strings.

I suggest that you diff with the browsermt:ug-bytearray-load branch (which is based on this PR) and merge the changes if you're happy with them.

**Comment by jerinphilip** [comment]
@ugermann Your commit is added here, and tests seem to be still passing. Are you still doing the string_view based hashmap mapped to a string_view value? In this case I'd have to keep AlignedMemory which is going to be used to construct the string_view alive at `Service`.

File: src/ssplit/ssplit.cpp (hunk 1 above)
Comment (ugermann) [reply src/ssplit/ssplit.cpp]: We removed the in-memory stream, so this bit of code has been removed. But as a general principle, I'm in favor of comments indicating end of namespace, as those scopes typically exceed one screen worth of code.

**Comment by ugermann** [comment]
@jerinphilip No, but there's still some typedef-ing going on to accommodate the gcc compilers on Valhalla that are too old to know about std::string_view. See src/ssplit/typedefs.h

**Review by ugermann — APPROVED** [review]

**Pull request merged**